### PR TITLE
w278177436 Remove default master branch parameter from git-scripts

### PR DESCRIPTION
**Pull request by jeet4320**
Closes https://www.wrike.com/open.htm?id=278177436
Fixes 
https://github.com/bodyshopbidsdotcom/git-scripts/issues/396
https://github.com/bodyshopbidsdotcom/git-scripts/issues/397
https://github.com/bodyshopbidsdotcom/git-scripts/issues/404

## What
Pull the default branch from GitHub API instead of `.git-scripts.yml` file 
## Why
Teams don't have to update their `.git-scripts.yml` file if they change default branch 
### How to reproduce error?
1. Checkout `release` branch of `git-scripts`
2. Run workon git-scripts and run pip install -r requirements.txt
3. Checkout `vice-frontend`, run `git_create_pr`. Note: `vice-frontend` default branch is release and it doesn't have any `.git-scripts.yml` file
4. You should see `GitCommandError` 

### Now how to test the fix?
1. Checkout `this` branch of `git-scripts`
2. Run `workon git-scripts` and run `pip install -r requirements.txt`
3. Checkout `vice-frontend`, run `git_create_pr`
4. You should not see any error 

Run below commands for regression:
1. `git_update_current`
2. `git_check_for_updates`
3. `git_create_pr`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,10 +5,10 @@ requests==2.9.1
...
 wsgiref==0.1.2
 PyWrike==0.5
-githubgateway==0.8
 PyRollbarAPI==0.7
 PyGithub==1.26.0
 inflection==0.3.1
 PyYAML==3.11
 awscli==1.15.63
 awsebcli==3.14.3
+-e git+git://github.com/bodyshopbidsdotcom/githubgateway.git@api-repo-info#egg=api-repo-info
```
Comment (jeet4320) [inline requirements.txt]: this will be updated once the this [PR](https://github.com/bodyshopbidsdotcom/githubgateway/pull/12) is in.

**Comment by whatisdot** [comment]
@jeet4320 Do you have an example of one repository that we can use to confirm the bug, and assert this resolves the issue?

**Comment by jeet4320** [comment]
@whatisdot update PR description with detailed steps on how to reproduce error and how to test the fix

File: requirements.txt (hunk 1 above)
Comment (whatisdot) [reply requirements.txt]: @jeet4320 You are good to merge githubgateway and update this to the next version.

Comment (jeet4320) [reply requirements.txt]: I would change this once this PR is good as based on comments I may have to change githubgateway PR.

Comment (santi-h) [reply requirements.txt]: @jeet4320 we need to publish the new version of `githubgateway`. Do you know how to do that?

Comment (jeet4320) [reply requirements.txt]: nope, guide me

File: lib/misc/helper2.py [hunk 2]
```diff
@@ -158,7 +158,8 @@ def reset_hard(self):
     self._git.reset('--hard')
 
   def base_branch(self):
-    return self.yaml('base_branch') or DEFAULT_BASE_BRANCH
+    self.github_api = GithubAPIGateway(*self.owner_and_repo(), token=Helper2.get_var('GITHUB_TOKEN'))
+    return  self.github_api.get_repo_details()['default_branch'] or DEFAULT_BASE_BRANCH
```
[the remark this answers is not in the document]
Comment (whatisdot) [reply lib/misc/helper2.py]: We also should only assign the value to `self.github_api` if it hasn't already been set.

[the remark this answers is not in the document]
Comment (jeet4320) [reply lib/misc/helper2.py]: Done

**Comment by jeet4320** [comment]
@santi-h can you please review this PR

File: lib/create_pr.py [hunk 3]
```diff
@@ -22,6 +22,8 @@
   unescape = HTMLParser().unescape
 
 def run():
+  if Helper2().uncommited_changes():
+    raise Exception("Please commit your changes or stash them before you deploy.")
```
Comment (santi-h) [inline lib/create_pr.py]: Why `"deploy"`? this script is for creating a PR

File: lib/misc/helper2.py [hunk 4]
```diff
@@ -158,7 +171,15 @@ def reset_hard(self):
...
 
   def base_branch(self):
-    return self.yaml('base_branch') or DEFAULT_BASE_BRANCH
+    base_branch = self.yaml('base_branch') or DEFAULT_BASE_BRANCH
+    try:
+      self._git.rev_parse(base_branch)
+    except Exception as e:
+      print(" Please make sure file '.git-scripts.yml' is present in your project folder and it has key 'base_branch' with correct default branch.")
+      print(" For example: base_branch: release")
+      sys.exit(1)
```
Comment (santi-h) [inline lib/misc/helper2.py]: Note that there's no other place in this file where we print anything out nor do we exit the scripts from here. We should keep it that way shouldn't have that kind of control. We should do something like what we did [here](https://github.com/bodyshopbidsdotcom/git-scripts/blob/master/lib/misc/helper2.py#L426), create a custom exception and let the client (in this case the actual scripts) handle it however they want

**Comment by jeet4320** [comment]
@santi-h update PR with comments

**Pull request merged**